zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -190,10 +210,30 @@ void closeAndRemoveInterpreterGroup(String noteId) {
     }
   }
 
+  void closeAndRemoveInterpreterGroupByUser(String user) {
+    if (user.equals("anonymous")) {
```

Comment:
I agree with @felixcheung and we also have to handle all of "anonymous" hardcoded. Actually, zeppelin-server module has a constant of "anonymous" thus we have to move it into another module like zeppelin-interpreter. I, however, it's not the scope of this PR.